src/mutations.ts

```diff
@@ -133,7 +133,25 @@ function applySchemaRules(
           applySchemaRules(propertySchema, values[key] as ObjectValue, options, jsonLogicContext)
         }
         if (propertySchema.items) {
-          applySchemaRules(propertySchema.items as JsfObjectSchema, values[key], options, jsonLogicContext)
+          if (propertySchema['x-jsf-presentation']?.inputType === 'group-array' && Array.isArray(values[key])) {
+            /*
+             * This is a partial workaround for group-array field items with conditional visibility rules.
+             * Due to the nature of group-array, the value is an array. applySchemaRules expects an object
+             * and it simply does not process the rules if the value is not an object.
+             *
+             * The correct solution would be to refactor applySchemaRules to handle arrays properly,
+             * but for now we simply pass an empty object to ensure the rules are applied.
+             *
+             * This means that the visibility rules for group-array items will only be based on the
+             * schema and will not work based on the actual values of the items in the array.
+             *
+             * This is not ideal, but it's better than the previous situation where the rules were not applied at all.
+             */
+            applySchemaRules(propertySchema.items as JsfObjectSchema, {}, options, jsonLogicContext)
+          }
+          else {
+            applySchemaRules(propertySchema.items as JsfObjectSchema, values[key], options, jsonLogicContext)
```

Comment:
Why do we need a if/else here? AFAIK every field with `items` is a group-array and the value it's an array too.

I also tried to run all the tests with this else commented out - and tests still passed. 

I'd appreciate @lukad eyes and expertise here to confirm my assumption

Reply:
So, passing a value as `{}` works because `applySchemaRules()` does a early returns if the value is not an array (which would always return falsy, even if the condition matches). 
As the `if` is hardcoded (a boolean true/false), the `validateSchema` doesn't even evaluate the value itself, regardless if it's `{}` or something else. It only looks at `typeof schema === 'boolean'`.

I believe the root bug is somewhere at `validateArray()`... still checking

Reply:
Which is basically what @gerardabello wrote in the comments, but it's hard to grasp until you really go through the code. I need to drop, this is where I'm left. I'd need considerable more time to debug this properly, which I can't right now.

Okay I wanna unblock you @gerardabello. Here's the plan:
1. Remove the redundant conditional
2. Write the extra test I asked
4. Open a MR in Dragon to ensure all tests still pass with a dev ver
5. We merge and release this

Thank you @gerardabello for your patience 🙇